controllers/api/api_trusted_controller.py

```diff
@@ -108,11 +109,16 @@ def _process_request(self, request, event_key):
 
             if (not match.time or match.time == "") and match.time_string:
                 # We can calculate the real time from the time string
-                logging.debug("Calculating time!")
-                MatchHelper.add_match_times(event, [match])
-
+                needs_time.append(match)
             matches.append(match)
 
+        if needs_time:
```

Comment:
Because I actually implemented it wrong the first time. That method should take a list of events, and it will start at the end of the list, and calculate times starting on the last day of the event and work backwards. So for multiday events, it would always think the matches are on the last day, since it will always assume the first match in the list takes place on the final day of the event.